onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -21,16 +21,31 @@ def compute_negative_log_likelihood_loss(input, target, weight=None, reduction='
         for i in range(N):
             if target[i] != ignore_index:
                 neg_gather_element_input[i] = -input[i][target[i]]
-    else:
+    elif len(input_shape) == 3:
```

Comment:
This may be a naive question: if we transpose the first two axes of input (so that C becomes the first axis) and flatten into a 2D (making all axes except first one into one axis), would it help simplify things? Of course, "target" also has to be similarly flattened. In other words, is there any difference in the treatment of axes other than C? If all those axes are treated the same way, it would simplify the logic to flatten them.

Reply:
At the least, "input = input.reshape((N,C,-1))" and "target = target.reshape((N,-1))" should let us handle different D cases uniformly.

Reply:
good point. I will try that.